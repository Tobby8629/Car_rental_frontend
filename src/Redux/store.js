import { configureStore } from '@reduxjs/toolkit';
import UserSlice from './UserSlice';
import CarSlice from './CarSlice';
import ReservationSlice from './ReservationSlice';

const store = configureStore({
  reducer: {
    Users: UserSlice,
    Cars: CarSlice,
    Reserve: ReservationSlice,
  },
});

export default store;
