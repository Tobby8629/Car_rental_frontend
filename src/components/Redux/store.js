import { configureStore } from '@reduxjs/toolkit';
import UserSlice from './UserSlice';
import ReservationSlice from './ReservationSlice';

const store = configureStore({
  reducer: {
    Users: UserSlice,
    Reserve: ReservationSlice,
  },
});

export default store;
