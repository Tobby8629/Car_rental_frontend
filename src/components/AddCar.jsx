import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { AddnewCar } from '../Redux/CarSlice';

const AddCar = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const storage = localStorage.getItem('token');
  const user = JSON.parse(storage);
  const [carDatas, setCarDatas] = useState({
    name: '',
    description: '',
    photo: '',
    price: '',
    model: '',
    user_id: user.id || null,
  });

  const [success, setSuccess] = useState('');
  const [error, setError] = useState('');

  const handleSubmit = async (e) => {
    e.preventDefault();
    if (!carDatas.name || !carDatas.price || !carDatas.model || !carDatas.photo) {
      setError('Please Fill All The Fields');
      setTimeout(() => {
        setError('');
      }, 3000);
      return;
    }
    const formData = new FormData();

    formData.append('car[name]', carDatas.name);
    formData.append('car[description]', carDatas.description);
    formData.append('car[photo]', carDatas.photo);
    formData.append('car[price]', carDatas.price);
    formData.append('car[model]', carDatas.model);
    formData.append('car[user_id]', carDatas.user_id);

    await dispatch(AddnewCar(formData));
    setCarDatas({
      name: '',
      description: '',
      photo: '',
      price: '',
      model: '',
    });
    setSuccess('Car Added Successfully');
    navigate('/');
  };

  const handleChange = (e) => {
    setCarDatas({ ...carDatas, [e.target.name]: e.target.value });
  };

  return (
    <section className="container w-100 d-flex justify-content-center align-items-center">
      <div className="row">
        <div className="col-md-10 w-100">
          {success && <div className="alert alert-success">{success}</div>}
          {error && <div className="alert alert-danger">{error}</div>}
          <h1 className="text-primary">Add New Car</h1>
          <form onSubmit={handleSubmit} className="w-100">
            <div className="mb-3">
              <label htmlFor="carname" className="form-label text-primary">
                Car Name
                <input
                  type="text"
                  name="name"
                  value={carDatas.name}
                  onChange={handleChange}
                  id="carname"
                  className="form-control"
                  placeholder="Car Name"
                />
              </label>
            </div>
            <div className="mb-3">
              <label htmlFor="carPrice" className="form-label text-primary">
                Car Price
                <input
                  type="number"
                  name="price"
                  value={carDatas.price}
                  onChange={handleChange}
                  className="form-control"
                  id="carPrice"
                  placeholder="Car Price"
                />
              </label>
            </div>

            <div className="mb-3">
              <label htmlFor="carmodel" className="form-label text-primary">
                Car Model

                <input
                  type="text"
                  name="model"
                  value={carDatas.model}
                  onChange={handleChange}
                  className="form-control"
                  id="carmodel"
                  placeholder="Car Model"
                />
              </label>
            </div>

            <div className="mb-3">
              <label htmlFor="carImage" className="form-label text-primary">
                Car Image

                <input
                  type="file"
                  accept="image/*"
                  name="photo"
                  onChange={(e) => {
                    const file = e.target.files[0];
                    setCarDatas({ ...carDatas, photo: file });
                  }}
                  className="form-control"
                  id="carImage"
                  placeholder="Enter your image url"
                />

              </label>
            </div>

            <div className="mb-3">
              <label htmlFor="carDescription" className="form-label text-primary">
                Car Description

                <textarea
                  name="description"
                  value={carDatas.description}
                  onChange={handleChange}
                  className="form-control"
                  id="carDescription"
                  rows="3"
                />
              </label>
            </div>
            <div className="mb-3">
              <input type="submit" className="btn btn-primary" value="Add Car" />
            </div>
          </form>
        </div>
      </div>
    </section>
  );
};

export default AddCar;
