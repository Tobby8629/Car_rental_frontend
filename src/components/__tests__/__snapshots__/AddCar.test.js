import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import AddCar from '../../AddCar';
import store from '../../../Redux/store';

test('AddCar component matches snapshot', () => {
  const user = { id: '123' }; // Set the desired user ID here
  const localStorageMock = {
    getItem: jest.fn().mockReturnValue(JSON.stringify(user)),
  };
  Object.defineProperty(window, 'localStorage', {
    value: localStorageMock,
    writable: true,
  });

  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <AddCar />
      </BrowserRouter>
    </Provider>,

  );
  expect(asFragment()).toMatchSnapshot();
});
