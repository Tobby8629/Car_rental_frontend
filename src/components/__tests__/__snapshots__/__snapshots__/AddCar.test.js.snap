// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCar component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="container w-100 d-flex justify-content-center align-items-center"
  >
    <div
      class="row"
    >
      <div
        class="col-md-10 w-100"
      >
        <h1
          class="text-primary"
        >
          Add New Car
        </h1>
        <form
          class="w-100"
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label text-primary"
              for="carname"
            >
              Car Name
              <input
                class="form-control"
                id="carname"
                name="name"
                placeholder="Car Name"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label text-primary"
              for="carPrice"
            >
              Car Price
              <input
                class="form-control"
                id="carPrice"
                name="price"
                placeholder="Car Price"
                type="number"
                value=""
              />
            </label>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label text-primary"
              for="carmodel"
            >
              Car Model
              <input
                class="form-control"
                id="carmodel"
                name="model"
                placeholder="Car Model"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label text-primary"
              for="carImage"
            >
              Car Image
              <input
                accept="image/*"
                class="form-control"
                id="carImage"
                name="photo"
                placeholder="Enter your image url"
                type="file"
              />
            </label>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label text-primary"
              for="carDescription"
            >
              Car Description
              <textarea
                class="form-control"
                id="carDescription"
                name="description"
                rows="3"
              />
            </label>
          </div>
          <div
            class="mb-3"
          >
            <input
              class="btn btn-primary"
              type="submit"
              value="Add Car"
            />
          </div>
        </form>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
